src/components/target-information-box.ts

```diff
@@ -12,22 +13,40 @@ class TargetInformationBoxElement extends LitElement {
 
   @state() _currentPosition: string = '';
   @state() _reversePosition: string = '';
+  @state() _currentMovement: number[][];
 
   connectedCallback() {
     super.connectedCallback();
   }
 
+  searchAddress(coordiante: number[]) {
+    SearchApi.getAddressFromCoordinate(coordiante).then((data) => {
+      this._reversePosition = data.results.length > 0 ? `proche de ${data.results[0].attributes.strname_deinr}` : 'Aucune adresse proche reconnue';
+    });
+  }
+
   constructor() {
     super();
+    this._currentMovement = [this.defaultPosition, this.defaultPosition];
+    const minCoordinateMoveBeforeSearchAddress = 20;
+
     window.addEventListener('current-center-position', ((event: CustomEvent) => {
-      this._reversePosition = useStore().getOptions().geolocationInformation.reverseLocation ? `${event.detail[0].toFixed(6)}}, ${event.detail[1].toFixed(6)}}` : '';
-      this._currentPosition = useStore().getOptions().geolocationInformation.currentLocation ? `${event.detail[0].toFixed(6)}}, ${event.detail[1].toFixed(6)}}` : '';
+      if (useStore().getOptions().geolocationInformation.reverseLocation) {
+        this._currentMovement[1] = event.detail;
+        if (Math.abs(this._currentMovement[0][0] - this._currentMovement[1][0]) > minCoordinateMoveBeforeSearchAddress || Math.abs(this._currentMovement[0][1] - this._currentMovement[1][1]) > minCoordinateMoveBeforeSearchAddress) {
```

Comment:
personally I think I would prefer to use lastPosition and newPosition than an array but you can keep the array if you think makes more sense

Reply:
Yes, the array value is clear like that.